pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -105,7 +106,45 @@
   @Override
   public void init(SegmentGeneratorConfig config)
       throws Exception {
-    init(config, RecordReaderFactory.getRecordReader(config));
+    init(config, getRecordReader(config));
+  }
+
+  private RecordReader getRecordReader(SegmentGeneratorConfig segmentGeneratorConfig)
+      throws Exception {
+    File dataFile = new File(segmentGeneratorConfig.getInputFilePath());
+    Preconditions.checkState(dataFile.exists(), "Input file: " + dataFile.getAbsolutePath() + " does not exist");
+
+    Schema schema = segmentGeneratorConfig.getSchema();
+    FileFormat fileFormat = segmentGeneratorConfig.getFormat();
+    String recordReaderPath = segmentGeneratorConfig.getRecordReaderPath();
```

Comment:
please rename this to recordReaderClassName, otherwise seems like a file path.

Reply:
Done